helm/thingsboard/templates/initializedb-job.yaml

```diff
@@ -44,11 +44,16 @@ spec:
           - mountPath: /config
             name: '{{ .Release.Name }}-node-config'
         image: "{{ .Values.node.image.server | default .Values.global.image.server }}/{{ .Values.node.image.repository }}:{{ .Values.node.image.tag | default .Values.global.image.tag }}"
-        command:
-          - "/bin/bash"
+        command: ["/bin/bash", "-c"]
         args:
-          - "-c"
-          - "start-tb-node.sh"
+          -|
+          start-tb-node.sh; \
+          export PGPASSWORD="{{ index .Values "postgresql-ha" "postgresql" "password" }}" \
+          psql -w -h "{{ include "thingsboard.pgpoolservicename" . }}" \
+          -p "{{ index .Values "postgresql-ha" "pgpool" "containerPorts" "postgresql" }}" \
+          -U "{{ index .Values "postgresql-ha" "postgresql" "username" }}" \
+          -d "{{ index .Values "postgresql-ha" "postgresql" "database" }}" \
+          -c "UPDATE queue SET partitions = 2;"
```

Comment:
[tb-node](https://github.com/midokura/thingsboard-ce-k8s/blob/master/helm/thingsboard/templates/node.yaml#L64-L68) has a `check-db-queue-ready` init-container that waits for the `queue` rows to be created before allowing TB node to start.
there's a obvious race-condition here, with this changing the partitions while tb node is booting up.

my suggestion is to make the "2" based on a configuration (wasn't one for partitions already?) and have the `check-db-queue-ready` validate the rows contain the expected value.
(this assumes no operator changes the value from the api or the db)

Reply:
Changes done! Let me know what you think I'm going to try to test it locally